### DISCUSSION TITLE
Simple Row splits

### DISCUSSION THREAD

**Discussion by SOTEL1** [jokergoo/ComplexHeatmap#791] [Q&A]
Hello ComplexHeatMap team, first of all thank you for this amazing package, and apologies for having difficulties with such simple procedures, despite your amazing tutorials --'. I will be basing my question on the examples and data in your tutorial :

https://jokergoo.github.io/ComplexHeatmap-reference/book/a-single-heatmap.html#heatmap-split

I am trying to split my heatmap based on specific rows, without any k-mean clustering. I have a set of data that I organized and would like to split my heatmap based on row numbers. However, I was not able to do so using the tutorial 

# split by a vector
Heatmap(mat, name = "mat", 
    row_split = rep(c("A", "B"), 9), column_split = rep(c("C", "D"), 12))

Say in the (mat) data I wish to split the HeatMap into 3 sections using "row_split" with one section containing rows (1:9) and the other two containing rows (10:14) and (15:18) respectively, how do I go about it? 

Also, what if I would want to make a simple annotation for rows, using ''HeatmapAnnotation(bar =...)'' whereby specific rows ((1:9) (10:14) (15:20)) each group having a specific value or letter ? 

Thank you for your time, 

Sincerely

**Comment by jokergoo** [comment]
Hi, you can set the `row_split` variable as follows:

```r
row_split = rep(c("A", "B", "C"), times = c(9, 4, 4))
Heatmap(..., row_split = row_split)
```

For the second question, since you are constructing the row annotation, you need to use `rowAnnotation()` function.
